slick/src/main/scala/slick/jdbc/HsqldbProfile.scala

```diff
@@ -125,8 +128,160 @@ trait HsqldbProfile extends JdbcProfile {
     }
     override val uuidJdbcType = new UUIDJdbcType {
       override def sqlType = java.sql.Types.BINARY
+
       override def sqlTypeName(sym: Option[FieldSymbol]) = "BINARY(16)"
     }
+    override val localTimeType = new LocalTimeJdbcType {
+      override def sqlType = java.sql.Types.TIME
+
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "TIME(3)"
+
+      override def getValue(r: ResultSet, idx: Int): LocalTime = {
+        // Fixes an issue caused because Hsqldb outputs the time in a non-standard format
+        // not adding trailing zeros to the hours. For example: '2:14:41.421' instead of '02:14:41.421'
+        r.getString(idx) match {
+          case null => null
+          case isoTime if isoTime.indexOf(':') == 2 => LocalTime.parse(isoTime)
```

Comment:
Why is the formatter on line 162 not used?

Reply:
The formatter on line 162 is for offset times. It has a non-optional offset component and so can't be used for LocalTime. The default LocalTime parser (which is used here) works fine though.